tests/model/SiteTreeTest.php

```diff
@@ -1226,3 +1260,28 @@ public function augmentValidURLSegment() {
 	}
 
 }
+
+class SiteTreeTest_SpecificParent1 extends Page implements TestOnly {
+	// Cannot have any parents at all.
+	private static $allowed_parents = array();
```

Comment:
I think we allow and test `$allowed_parents = 'none'`. We added an explicit non-falsey 'none' value in order to overcome the behaviour inherit in config at not being good at distinguishing falsey values.

SiteTree adds a default `private static $allowed_children = array("SiteTree");` so perhaps we should be consistent and add the same for allowed_parents?

This would mean your code would always expect a value for allowed_parents, rather than treating empty values as an implicit 'None', and reduce the risk of getting the 'wrong' empty values for root-only pages. :)